### ISSUE TITLE
Polyhedron demo regression: loading a polygon soup can segfault (was working in 4.6)

### ISSUE THREAD

**Issue by lrineau** [CGAL/cgal#299]
@sloriot: In CGAL-4.7, the loading of a polygon soup as a STL file can crash.

[Example file](https://geometryfactory.codebasehq.com/upload/2afdaecd-c6cd-4425-011e-53d282e9929c/show/original), unfortunately only for GeometryFactory's employees.

The code is:
```C++
  try{
    // Try building a polyhedron
    Polyhedron P;
    CGAL::Polygon_mesh_processing::polygon_soup_to_polygon_mesh(points, triangles, P);
    
    if(! P.is_valid() || P.empty()){
      std::cerr << "Error: Invalid polyhedron" << std::endl;
    }
    else{
      Scene_polyhedron_item* item = new Scene_polyhedron_item(P);
      item->setName(fileinfo.baseName());
      return item;
    }
  }
  catch(...){}

  Scene_polygon_soup_item* item = new Scene_polygon_soup_item();
  item->setName(fileinfo.baseName());
  item->load(points, triangles);
  return item;
```

In CGAL-4.6, the loading of a polygon soup to a polyhedron was different:
```C++
  try{
    // Try building a polyhedron
    Polyhedron P;
    CGAL::polygon_soup_to_polyhedron_3(P, points, triangles);
[...]
  }
  catch(...){}
```
`CGAL::polygon_soup_to_polyhedron_3` uses the polyhedron incremental builder, where as `CGAL::Polygon_mesh_processing::polygon_soup_to_polygon_mesh` uses BGL code.

The incremental builder was able to detect all errors, whereas the new code just segfault!

**Comment by lrineau** [comment]
@sloriot, do you have an update on that issue?

**Comment by sloriot** [comment]
See #312

**Comment by lrineau** [comment]
Fixed by #312
